### PR TITLE
Add docstrings to predict functions and normalize to Google docstring format

### PR DESCRIPTION
**Pull request by j201**
Includes removing redundant type annotations given in function signatures

### CODE REVIEW

File: src/tabdpt/classifier.py [hunk 1]
```diff
@@ -81,7 +81,24 @@ def predict_proba(
...
+
+        Args:
+            X: Input inference instances, n_instances x n_features.
+            temperature: Temperature scaling applied before softmax, can be used to calibrate probability outputs.
+            context_size: Maximum number of train points in the context. Uses all points if `None`, which can lead to
+                GPU OOMs. Otherwise reduces context size based on `context_reduction` setting.
+            batch_size: Number of inference points to use in each batch. If `None`, defaults to 4096 on CPU and 128k on
+                GPU. Can be increased for faster inference, or decreased to prevent OOMs.
+            return_logits: If True, returns logits without temperature scaling or softmax.
+            seed: Seed used for permuting feature/class order during ensembling. If n_ensembles is 1, then feature
```
Comment (anthonycaterini) [inline src/tabdpt/classifier.py]: nit: should have backticks around `n_ensembles`

File: src/tabdpt/classifier.py [hunk 2]
```diff
@@ -81,7 +81,24 @@ def predict_proba(
         return_logits: bool = False,
         seed: int | None = None,
         class_perm: np.ndarray | None = None,
-    ):
+    ) -> np.ndarray:
+        """Predict classification output without ensembling, returning class probabilities
+
+        Args:
+            X: Input inference instances, n_instances x n_features.
+            temperature: Temperature scaling applied before softmax, can be used to calibrate probability outputs.
```
Comment (anthonycaterini) [inline src/tabdpt/classifier.py:89]: "probabilistic" outputs? 
or "probability" outputs?

File: src/tabdpt/classifier.py [hunk 3]
```diff
@@ -81,7 +81,24 @@ def predict_proba(
         return_logits: bool = False,
         seed: int | None = None,
         class_perm: np.ndarray | None = None,
-    ):
+    ) -> np.ndarray:
+        """Predict classification output without ensembling, returning class probabilities
+
+        Args:
+            X: Input inference instances, n_instances x n_features.
```
Comment (anthonycaterini) [inline src/tabdpt/classifier.py]: we have a bunch of different nomenclature in this file. it may be worth standardizing at some point, but then we will lose backward compatibility. possibly a TODO, or possibly we don't care. just highlighting

case in point: I have seen `n_samples`, `num_features`, `num_classes`, `n_instances`, `n_features`, `n_ensembles`. 

at first I was thinking we should map the docstring to the actual values in the code and add backticks (e.g., n_features -> `num_features`), but there is no `n_instances` in the actual code anywhere.

so I don't have a concrete recommendation really, just bringing this to light

File: src/tabdpt/classifier.py [hunk 4]
```diff
@@ -81,7 +81,24 @@ def predict_proba(
...
+            temperature: Temperature scaling applied before softmax, can be used to calibrate probability outputs.
+            context_size: Maximum number of train points in the context. Uses all points if `None`, which can lead to
+                GPU OOMs. Otherwise reduces context size based on `context_reduction` setting.
+            batch_size: Number of inference points to use in each batch. If `None`, defaults to 4096 on CPU and 128k on
+                GPU. Can be increased for faster inference, or decreased to prevent OOMs.
+            return_logits: If True, returns logits without temperature scaling or softmax.
```
Comment (anthonycaterini) [inline src/tabdpt/classifier.py]: nit: backticks on True?

File: src/tabdpt/classifier.py [hunk 5]
```diff
@@ -81,7 +81,24 @@ def predict_proba(
...
+            seed: Seed used for permuting feature/class order during ensembling. If n_ensembles is 1, then feature
+                permutation will only be done if this is set to a non-`None` value.
+            class_perm: An optional permutation of classes to apply before prediction, mainly for ensembling.
+
+        Returns:
+            A class probability array of size n_instances x n_classes.
```
Comment (anthonycaterini) [inline src/tabdpt/classifier.py]: backticks here? What about `num_classes` ?

File: src/tabdpt/model.py [hunk 6]
```diff
@@ -84,10 +84,12 @@ def forward(
         num_features: torch.Tensor,
     ) -> torch.Tensor:
         """Forward pass of the TabDPTModel.
+
         Args:
-            x_src (torch.Tensor): Input features of shape (B, T, F).
-            y_src (torch.Tensor): Target values of shape (B, T).
-            return_log_act_norms (bool): Whether to return activation norms for logging.
+            x_src: Input features of shape (B, T, F).
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: what is `T` meant to represent?

File: src/tabdpt/model.py [hunk 7]
```diff
@@ -84,10 +84,12 @@ def forward(
...
+            y_src: Target values of shape (B, T).
+            num_features: Number of features
+
         Returns:
             torch.Tensor: Predicted values of shape (T, B, n_out + regression_bin_count).
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: just confirming that this predicted shape is correct? Should we have less stuff in the predicted output modulated by `eval_pos` ?

File: src/tabdpt/model.py [hunk 8]
```diff
@@ -84,10 +84,12 @@ def forward(
...
     ) -> torch.Tensor:
         """Forward pass of the TabDPTModel.
+
         Args:
-            x_src (torch.Tensor): Input features of shape (B, T, F).
-            y_src (torch.Tensor): Target values of shape (B, T).
-            return_log_act_norms (bool): Whether to return activation norms for logging.
+            x_src: Input features of shape (B, T, F).
+            y_src: Target values of shape (B, T).
+            num_features: Number of features
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: `num_features` is not used in this function

File: src/tabdpt/model.py
Comment (anthonycaterini) [inline src/tabdpt/model.py:1]: FYI, `enc_cell_dim` and `num_col_attn_layers` are not used in the `__init__` function

File: src/tabdpt/model.py [hunk 9]
```diff
@@ -187,14 +189,15 @@ def __init__(
...
-            num_heads (int): Number of attention heads.
-            ff_dim (int): Dimension of the feed-forward network.
-            base_len (int): Base length for attention scaling.
-            max_len (int): Maximum length for attention scaling. If equal to base_len, attention scaling is disabled.
-            y_encoder_dim (int): Dimension of per-layer y embedding; v_proj input is embed_dim + y_encoder_dim.
+            embed_dim: Dimension of the embedding.
+            num_heads: Number of attention heads.
+            ff_dim: Dimension of the feed-forward network.
+            base_len: Base length for attention scaling.
+            max_len: Maximum length for attention scaling. If equal to base_len, attention scaling is disabled.
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: nit: backticks

File: src/tabdpt/model.py [hunk 10]
```diff
@@ -187,14 +189,15 @@ def __init__(
...
+            y_encoder_dim: Dimension of per-layer y embedding; v_proj input is embed_dim + y_encoder_dim.
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: nit: backticks

File: src/tabdpt/model.py [hunk 11]
```diff
@@ -243,9 +246,10 @@ def forward(self, x: torch.Tensor, y: torch.Tensor, eval_pos: int) -> torch.Tens
         """Compute the residual for this layer.
 
         Args:
-            x (torch.tensor): Input tensor of shape (L, B, D).
-            y (torch.tensor): Target embedding for the context region of shape (eval_pos, B, y_encoder_dim).
-            eval_pos (int): Number of context positions (length of y and K/V).
+            x: Input tensor of shape (L, B, D).
+            y: Target embedding for the context region of shape (eval_pos, B, y_encoder_dim).
```
Comment (anthonycaterini) [inline src/tabdpt/model.py]: nit: backticks

**Review by anthonycaterini — CHANGES_REQUESTED** [review]
Various minor changes. I didn't repeat comments from the first shared docstring across all docstrings with the same issues. One of the biggest things was the use of backticks in docstrings around variables in the description

File: src/tabdpt/classifier.py (hunk 2 above)
Comment (j201) [reply src/tabdpt/classifier.py:89]: Either, I meant outputs from `predict_proba` and `ensemble_predict_proba` that are probabilities.

File: src/tabdpt/model.py (hunk 8 above)
Comment (j201) [reply src/tabdpt/model.py]: It isn't, but I didn't address in this commit because I wanted to avoid any functional changes to the code

File: src/tabdpt/classifier.py (hunk 3 above)
Comment (j201) [reply src/tabdpt/classifier.py]: I agree, based on current usage, I would suggest `n_features`, `n_instances`, `n_classes`, `n_ensembles`. I would also prefer to use more informative terms in comments specifying dimensions. E.g., in `model.py` we have dimensions called things like `T` and `L` where I always have to read the code to remind myself what they actually are.

File: src/tabdpt/model.py (hunk 6 above)
Comment (j201) [reply src/tabdpt/model.py]: Yeah I always have to remind myself, but it's throughout the code. `B` is the normal batching dimension where each slice is operated on independently. `T` is the transformer sequence dimension. So when we do TabPFN-style batching, we actually concatenate everything across the `T` dimension and `B == 1`.

File: src/tabdpt/model.py (hunk 7 above)
Comment (j201) [reply src/tabdpt/model.py]: I think this comment was originally meant to give the meaning of each dimension rather than the tensor sizes, and got muddled later. For instance, `x_src` is longer in the `T` dimension than `y_src` because it includes eval samples. I'll try a rewrite.

**Comment by j201** [comment]
Added backticks throughout and improved the `model.forward` documentation to clarify.

File: src/tabdpt/classifier.py (hunk 3 above)
Comment (anthonycaterini) [reply src/tabdpt/classifier.py]: ok let's put a pin in this. should we add an explicit TODO or an Issue?

File: src/tabdpt/model.py
Comment (anthonycaterini) [reply src/tabdpt/model.py:1]: Issue or TODO?

**Review by anthonycaterini — APPROVED** [review]

File: src/tabdpt/classifier.py (hunk 3 above)
Comment (j201) [reply src/tabdpt/classifier.py]: Maybe just something to keep in mind? Not sure how to summarize this into a tidy group of changes, but I did add some more documentation for `T` at least.

File: src/tabdpt/model.py
Comment (j201) [reply src/tabdpt/model.py:1]: These parameters are used in some internal code branches that aren't in this repo, they ended up here when porting over changes, I think it's fair to keep them around to limit the difference between code versions.
